sonar-text-plugin/src/main/java/org/sonar/plugins/secrets/configuration/SchemaValidator.java

```
@@ -0,0 +1,78 @@
+import com.networknt.schema.ValidationMessage;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InvalidObjectException;
+import java.net.URL;
+import java.util.Set;
+
+public class SchemaValidator {
+
+  private static final String FILE_LOCATION = "/org/sonar/plugins/secrets/configuration/";
+  private static final String VALIDATION_SCHEMA_FILE = "specification-json-schema.json";
+  private static final Set<String> SPECIFICATION_FILES = Set.of("minSpec.yaml");
+
+  private static final ObjectMapper MAPPER = new ObjectMapper(new YAMLFactory());
+
+  private static final JsonSchema VALIDATION_SCHEMA = initializeValidationSchema();
+
+  private SchemaValidator() {
+
+  }
+
+  private static JsonSchema initializeValidationSchema() {
+    JsonSchemaFactory schemaFactory = JsonSchemaFactory
+      .builder(JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V202012))
+      .objectMapper(MAPPER)
+      .build();
+    InputStream validationSchema = SchemaValidator.class.getResourceAsStream(FILE_LOCATION + VALIDATION_SCHEMA_FILE);
+    return schemaFactory.getSchema(validationSchema);
+  }
+
+  public static void validateConfigurationFiles() {
+    for (String configurationFile : SPECIFICATION_FILES) {
+      URL configuration = SchemaValidator.class.getResource(FILE_LOCATION + configurationFile);
+      validate(configuration);
+    }
+  }
+
+  public static void validate(URL configurationLocation) {
+    try {
+      Set<ValidationMessage> validate = VALIDATION_SCHEMA.validate(MAPPER.readTree(configurationLocation));
+      if (!validate.isEmpty()) {
+        String filePath = configurationLocation.getPath();
+        String fileName = filePath.substring(filePath.lastIndexOf('/') + 1);
+        String errorMessage = String.format("Specification file \"%s\" failed the schema validation", fileName);
+        throw new InvalidObjectException(errorMessage);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
```

You should throw a custom Exception like SchemaValidationException